internal/storage/db/internal/db.go

```diff
@@ -655,14 +676,63 @@ func (s *dbStorage) ListPolicyIDs(ctx context.Context, includeDisabled bool) ([]
 		return nil, fmt.Errorf("could not execute %q query: %w", "ListPolicyIDs", err)
 	}
 
-	policyIDs := make([]string, len(policyCoords))
-	for i := 0; i < len(policyCoords); i++ {
-		policyIDs[i] = policyCoords[i].PolicyKey()
+	policyIDs := make([]string, 0, len(policyCoords))
+	for _, pc := range policyCoords {
+		if checkPostFilters(pc, postFilters) {
+			policyIDs = append(policyIDs, pc.PolicyKey())
+		}
 	}
 
 	return policyIDs, nil
 }
 
+type postRegexpFilter struct {
+	re  *regexp.Regexp
+	col string
+}
+
+// updateRegexpFilters updates either `whereExprs` or `postFilters` in place, dependent on whether regexp support is enabled or not.
+func (s *dbStorage) updateRegexpFilters(namePattern, col string, whereExprs *[]exp.Expression, postFilters *[]postRegexpFilter) error {
+	r, err := s.opts.regexpCache.GetCompiledExpr(namePattern)
+	if err != nil {
+		return err
+	}
+	if s.regexpEnabled() {
+		// We need to pass a *regexp.Regexp expression to `Like` (or `RegexpLike`, which is equivalent) in order for goqu
+		// to correctly parse the query. We need to pass the compiled expression in order for goqu to access (only) the
+		// raw string (https://github.com/doug-martin/goqu/blob/master/exp/bool.go#L148).
+		// We use a cache to prevent the need to recompile arbitrary strings on each request.
+		// In the case of the SQLite driver, to support regexp, we generate an application-defined function in which we
+		// use the cached compiled expressions.
+		*whereExprs = append(*whereExprs, goqu.C(col).ILike(r))
+	} else {
+		*postFilters = append(*postFilters, postRegexpFilter{re: r, col: col})
+	}
+
+	return nil
+}
+
+func (s *dbStorage) regexpEnabled() bool {
+	// TODO(saml) link this to the `dialect` arg passed to `goqu` in the sqlserver package, or rethink how to indicate regexp support
+	return s.db.Dialect() != "sqlserver"
+}
+
+func checkPostFilters(pc namer.PolicyCoords, postFilters []postRegexpFilter) bool {
+	for _, f := range postFilters {
+		switch f.col {
+		case PolicyTblNameCol:
+			if !f.re.MatchString(pc.Name) {
+				return false
+			}
+		case PolicyTblScopeCol:
+			if !f.re.MatchString(pc.Scope) {
+				return false
+			}
```

Comment:
Should this include the version columns as well?

Reply:
🙈 thank you

Reply:
my tests are certainly not as good as they should be